internal/resource/discovery.go

```diff
@@ -0,0 +1,89 @@
+package resource
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"regexp"
+	"strconv"
+
+	"github.com/cybozu-go/necoperf/internal/constants"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	port = 6543
```

Comment:
The port number is configurable at the server-side.
The client should also be configurable.

The default number should be shared by the server and the client.